Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Undo/PrefabUndoEntityOverrides.cpp

```diff
@@ -0,0 +1,145 @@
+                return;
+            }
+
+            PrefabDomReference cachedOwningInstanceDom = owningInstance.GetCachedInstanceDom();
+
+            const AZStd::string overridePatchPathToOwningInstance = PrefabInstanceUtils::GetRelativePathFromClimbedInstances(
+                climbUpResult.m_climbedInstances, true); // skip top instance
+
+            for (const AZ::Entity* entity : entityList)
+            {
+                if (!entity)
+                {
+                    AZ_Error("Prefab", false, "PrefabUndoEntityOverrides::Capture - Found a null entity. Skipping it.");
+                    continue;
+                }
+
+                // Hierarchy Example: Level (focused) -> Car (top) -> ... -> Wheel (owning) -> Tire (override)
+                // Path seen from focused: /Instances/Instance_[Car]/.../Instances/Instance_[Wheel]/Entities/Entity_[Tire]
+                //                                   (top instance)               (owning instance)         (entity to update)
+                // Note: Override patch path is the path seen from top without top instance alias path.
+                const AZStd::string entityAliasPath = m_instanceToTemplateInterface->GenerateEntityAliasPath(entity->GetId());
+                const AZStd::string entityPathFromTopInstance = overridePatchPathToOwningInstance + entityAliasPath;
+                PrefabDomPath entityDomPathFromTopInstance(entityPathFromTopInstance.c_str());
+
+                // Get the after-state entity DOM.
+                PrefabDom entityDomAfterUpdate;
+                m_instanceToTemplateInterface->GenerateDomForEntity(entityDomAfterUpdate, *entity);
+
+                // Get the before-state entity DOM inside the source template DOM of the top instance.
+                const TemplateId topTemplateId = link->get().GetSourceTemplateId();
+                const PrefabDom& topTemplateDom = m_prefabSystemComponentInterface->FindTemplateDom(topTemplateId);
+
+                PrefabDom overridePatches;
+                {
+                    // This scope is added to limit their usage and ensure DOM is not modified when it is being used.
+                    // DOM value pointers can't be relied upon if the original DOM gets modified after pointer creation.
+                    const PrefabDomValue* entityDomInTopTemplate = entityDomPathFromTopInstance.Get(topTemplateDom);
+
+                    if (entityDomInTopTemplate)
+                    {
+                        m_instanceToTemplateInterface->GeneratePatch(overridePatches, *entityDomInTopTemplate, entityDomAfterUpdate);
+                    }
+                    else
+                    {
+                        // Merge patches to an entity DOM as add-entity override.
+                        // Path is empty because there is no subpath for add-entity override patch like "/Components/Component_[123]".
```

Comment:
comment is not correct because it'll enter the else path if it can't find the entity in the dom and entity path stops at '/Instances/..../Entities/someAlias'

Reply:
Also, when does this even happen?(Because the add entity nodes are supposed to take care of additions right? )

Regardless, a unit test for this case would be great.

Reply:
> Also, when does this even happen?(Because the add entity nodes are supposed to take care of additions right? )
> Regardless, a unit test for this case would be great.

So, this "else" branch is for covering the case that there is already an add-entity DOM as override, but the users continue editing the entity (like a component update) so these new override patches should be "merged" into that add-entity DOM.

Yeah a unit test for this is definitely needed.

Reply:
> comment is not correct because it'll enter the else path if it can't find the entity in the dom and entity path stops at '/Instances/..../Entities/someAlias'

I think you're referring to the entity path but my comment is referring to the empty path "" below. I need to update the comment for clearer description.